### PR TITLE
Add variables required to TCP direct connections

### PR DESCRIPTION
**Pull request by ktaborski**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: plugin/src/main/java/jenkins/plugins/openstack/compute/UserDataVariableResolver.java [hunk 1]
```diff
@@ -77,6 +80,21 @@
                 "Labels of the node.",
                 r -> r.labelString
         );
+        stub(
+                "AGENT_NAME",
+                "Agent name",
+                r -> r.serverName
+        );
+        stub(
+                "JENKINS_INSTANCE_IDENTITY",
```
Comment (olivergondza) [inline plugin/src/main/java/jenkins/plugins/openstack/compute/UserDataVariableResolver.java:89]: Question: What is the use-case for accessing instance identity?

Comment (ktaborski) [reply plugin/src/main/java/jenkins/plugins/openstack/compute/UserDataVariableResolver.java:89]: Please look on:
https://github.com/jenkinsci/remoting/blob/master/docs/inbound-agent.md#connect-directly-to-tcp-port
It is one of arguments.

Comment (olivergondza) [reply plugin/src/main/java/jenkins/plugins/openstack/compute/UserDataVariableResolver.java:89]: Thanks. Makes sense.

File: plugin/src/main/java/jenkins/plugins/openstack/compute/UserDataVariableResolver.java [hunk 2]
```diff
@@ -77,6 +80,21 @@
...
+                "Jenkins instance identity. See <a href='https://wiki.jenkins.io/display/JENKINS/Instance+Identity'>Instance Identity</a>",
+                r -> Util.fixNull(hudson.remoting.Base64.encode(org.jenkinsci.main.modules.instance_identity.InstanceIdentity.get().getPublic().getEncoded()))
```
Comment (olivergondza) [inline plugin/src/main/java/jenkins/plugins/openstack/compute/UserDataVariableResolver.java:91]: Please use `java.util.Base64.getEncoder()` as this class is deprecated.

**Comment by olivergondza** [comment]
Thanks for the contribution. Please address the minor comment and this is good to get merged.

**Comment by ktaborski** [comment]
What are plans to release new plugin version?

**Pull request merged**